deer/maths.py

```diff
@@ -33,6 +33,26 @@ def matmul_recursive(mats: jnp.ndarray, vecs: jnp.ndarray, y0: jnp.ndarray) -> j
     -------
     result: jnp.ndarray
         The result of the matrix multiplication, shape ``(nsamples, ny)``, including ``y0`` at the beginning.
+    
+    Example
+    -------
+    >>> import jax.numpy as jnp
+    >>> mats = jnp.array([
+    ...     [[1.0, 0.0],
+    ...      [0.0, 1.0]],
+    ...     [[0.5, 0.5],
+    ...      [0.5, 0.5]]
+    ... ])
+    >>> vecs = jnp.array([
+    ...     [1.0, 1.0],
+    ...     [0.0, 0.0]
+    ... ])
+    >>> y0 = jnp.array([1.0, 2.0])
+    >>> matmul_recursive(mats, vecs, y0)
+    Array([[1. , 2. ],
+           [2. , 3. ],
+           [2.5, 2.5]], dtype=float64)
+
```

Comment:
This is an internal function, so we don't need an example here

Reply:
It has been removed